sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/experimental/ViewportPlugin.kt

```diff
@@ -0,0 +1,154 @@
+package com.mapbox.maps.plugin.viewport.experimental
+
+import com.mapbox.maps.plugin.MapPlugin
+import com.mapbox.maps.plugin.viewport.experimental.data.DefaultViewportTransitionOptions
+import com.mapbox.maps.plugin.viewport.experimental.data.FollowingViewportStateOptions
+import com.mapbox.maps.plugin.viewport.experimental.data.OverviewViewportStateOptions
+import com.mapbox.maps.plugin.viewport.experimental.state.FollowingViewportState
+import com.mapbox.maps.plugin.viewport.experimental.state.OverviewViewportState
+import com.mapbox.maps.plugin.viewport.experimental.state.ViewportState
+import com.mapbox.maps.plugin.viewport.experimental.transition.ViewportTransition
+
+/**
+ * The Viewport plugin allows to track objects on a map.
+ *
+ * It provides a structured approach to organizing camera management logic into states and transitions between them.
+ *
+ * at any given time, the viewport is either:
+ *  - idle (not updating the camera)
+ *  - in a state (camera is being managed by a ViewportState)
+ *  - transitioning (camera is being managed by a ViewportTransition)
+ */
+interface ViewportPlugin : MapPlugin {
+
+  /**
+   * Returns current [ViewportStatus].
+   *
+   * If current status is IDLE, returns null.
+   * @see addStatusObserver
+   */
+  val status: ViewportStatus?
```

Comment:
Need to discuss optionality and idle representation here. On iOS, I updated this to be non-optional and made the ViewportState within the "State" case optional such that idle is expressed as `State(nil)`

Reply:
that works as well, I think it makes more sense, will refactor to use `State(null)`

Reply:
done